packages/zone/src/virtual.js

```diff
@@ -79,6 +83,7 @@ export const makeVirtualZone = (baseLabel = 'virtualZone') => {
     subZone: wrapProvider(makeSubZone),
 
     makeOnce,
+    watchPromise,
```

Comment:
We probably should put as a caveat that this implementation is not "virtual" in the sense that it will have a heap cost keeping the watcher in the heap. In theory this should use the liveslots provided `watchPromise`, but there seems to be a bug there where providing a virtual watcher is not in fact supported.